docs/modules/data-structures/pages/vector-collections.adoc

```diff
@@ -0,0 +1,472 @@
+CompletionStage<Void> resultDelete = vectorCollection.deleteAsync("1");
+CompletionStage<VectorDocument<String>> resultRemove = vectorCollection.removeAsync("2");
+----
+--
+====
+
+NOTE: These methods do not delete index nodes but mark them as deleted. This may impact search speed and memory usage. To permanently remove nodes from the index, you must run index optimization after deletion. See <<optimize-collection, optimize method>>
+
+== Similarity search
+
+Vector search returns entries with vectors that are most similar to the query vector, based on specified metrics. Any query consists of exactly one vector to search and search options, such as the limit of results to retrieve, etc. For more details about available options, refer to <<similarity-search-options, Similarity search options>>
+
+For similarity search, use the `searchAsync` method of the `VectorCollection`.
+
+In a single index vector collection, specifying the name of the index to search is not required.
+However, for a multi-index vector collection, specifying the name of the index to search in is necessary.
+
+Example for single-index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of(new float[] {0, 0}),
+        SearchOptions.builder()
+            .limit(10)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+Example for multi-index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of("index_name", new float[] {0, 0}),
+        SearchOptions.builder()
+            .limit(10)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Similarity search options
+Search parameters are passed as an argument `searchOptions` to the searchAsync method.
+
+.Search options
+[cols="1,2,1",options="header"]
+|===
+|Option|Description|Default
+
+|limit
+|How many results should be returned in search result
+|`1`
+
+|includeValue
+|Include the user value in the search result.
+|`N`
+
+
+|includeVectors
+|Include the vectors values in the search result.
+|`N`
+
+|hints
+|Extra hints for the search.
+|`None`
+
+|===
+
+
+.Available hints
+[cols="1,2",options="header"]
+|===
+|Hint|Description
+
+|Hints.PARTITION_LIMIT
+|Number of results to fetch from partition.
+
+|Hints.MEMBER_LIMIT
+|Number of results to fetch from member in 2-stage search.
+
+|Hints.FORCE_SINGLE_STAGE_SEARCH
+|Force use of 1-stage search.
+
+|===
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder()
+                .limit(10)
+                .includeValue()
+                .includeVectors()
+                .hint(Hints.PARTITION_LIMIT, 1)
+                .build();
+----
+--
+====
+
+=== 1-stage and 2-stage search
+...
```

Comment:
something missing here?

Reply:
description of 1-stage and 2 stage search

Reply:
deleted this block